### PR TITLE
1.Console-EchoBot for NODEJS

### PR DESCRIPTION
**Pull request by benbrown**
A simple console-based echobot example.

Even with this simple example, I thought we should start with the folder and file structure in place. Is it strictly necessary to pull the echo feature into the mainDialog? No, but its the right model.

Additionally, I put in the .env loader, even though it isn't yet used.

- [ ] I need help with the deploymentScripts.
- [ ] There is a link to docs for TurnContext objects that I copied from an old sample that seems broken.

### CODE REVIEW

File: javascript_nodejs/1.console-echobot/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "console-echobot",
+  "version": "1.0.0",
+  "description": "Bot Builder v4 console echobot sample",
+  "main": "index.js",
+  "scripts": {
+    "start": "node ./index.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (vishwacsena) [inline javascript_nodejs/1.console-echobot/package.json]: For consistency, we should add "watch": "nodemon ./index.js"

File: javascript_nodejs/1.console-echobot/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+// Licensed under the MIT License.
+
+const botbuilder = require('botbuilder');
+const path = require('path');
+const config = require('dotenv').config;
+
+const mainDialog = require('./dialogs/mainDialog');
+
+// load environment variables from .env file.
+const ENV_FILE = path.join(__dirname, '..', '.env');
```
Comment (vishwacsena) [inline javascript_nodejs/1.console-echobot/index.js]: Your .env file would typically be at root so unclear if we need '..' here. Would be better with path.join(__dirname, '.env');

File: javascript_nodejs/1.console-echobot/dialogs/mainDialog/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+module.exports = async function(context) {
+    // check to see if this activity is an incoming message
+    // (it could theoretically be another type of activity)
+    if (context.activity.type === 'message') {
+
+        // check to see if the user sent a simple "quit" message
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/dialogs/mainDialog/index.js]: Extra blank line

File: javascript_nodejs/1.console-echobot/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+After running `npm start`, the bot will present a prompt directly in the terminal window.
+
+Send messages to the bot by typing them into the terminal.  The bot will echo your message back to you!
+
+# Adapters
+
+Adapters such as the `ConsoleAdapter` provide an abstraction for your bot to work with a variety of environments and messaging platforms.
+
+The adapter is responsible for directing incoming and outgoing communication, authentication, and so on. Adapters for different platforms and
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/README.md]: "and so on" isn't a good way to end a sentence.

File: javascript_nodejs/1.console-echobot/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+messaging technologies differ internally, but achieve the same goal - connecting bots to a stream of messages and events from users. 
+When your bot receives an activity, the adapter wraps up everything about that activity, creates a [context object](https://github.com/Microsoft/botbuilder-js/blob/master/doc/botbuilder/classes/botbuilder.turncontext.md), passes it to your bot's application logic,
+and sends responses generated by your bot back to the user's channel.
+
+In most situations, we don't work with the adapter directly, but it's good to know it's there and what it does. In this example, we're using the
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/README.md]: who is "we" in this case? Consider removing or changing the articles.

Comment (cleemullins) [reply javascript_nodejs/1.console-echobot/README.md]: Strongly consider removing the entire first sentence in this paragraph.

File: javascript_nodejs/1.console-echobot/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+`ConsoleAdapter`, which provides a very simple way to exchange messages - they're sent and received via the terminal window. Using the ConsoleAdapter
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/README.md]: "very simple" or "simple". No need for the "very". simplify.

Comment (cleemullins) [reply javascript_nodejs/1.console-echobot/README.md]: "console window"

File: javascript_nodejs/1.console-echobot/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+is a great way of getting started quickly with Botbuilder. It also allows you to build and test bots that operate on your local machine without any
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/README.md]: BotBuilder vs Botbuilder.

Comment (cleemullins) [reply javascript_nodejs/1.console-echobot/README.md]: "It also" isn't a good way to start a sentence. Consider simplifying the grammar.

File: javascript_nodejs/1.console-echobot/README.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+external API calls or client applications.
+
+Other available adapters can connect your bot to the web, or through the BotFramework Connector Service, to many popular messaging platforms.
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/README.md]: remove "can". Simplify language.

File: javascript_nodejs/1.console-echobot/index.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+const loadFromEnv = config({path: ENV_FILE});
+
+// create the Bot adapter, which is responsible for sending and receiving messages
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/index.js]: "Create" vs "create"

Comment (cleemullins) [reply javascript_nodejs/1.console-echobot/index.js]: add a "." at the end.

File: javascript_nodejs/1.console-echobot/index.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+// in this case, we are using the ConsoleAdapter, which enables a bot you can chat with from within your terminal window.
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/index.js]: remove "in this case". Simplify language.

File: javascript_nodejs/1.console-echobot/index.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+const adapter = new botbuilder.ConsoleAdapter();
+
+// tell the adapter to start listening for incoming messages and events, known as "activities"
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/index.js]: Make these sentences.

File: javascript_nodejs/1.console-echobot/package.json [hunk 12]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "console-echobot",
+  "version": "1.0.0",
+  "description": "Bot Builder v4 console echobot sample",
```
Comment (cleemullins) [inline javascript_nodejs/1.console-echobot/package.json]: BotBuilder vs Bot Builder

File: javascript_nodejs/1.console-echobot/package.json (hunk 1 above)
Comment (benbrown) [reply javascript_nodejs/1.console-echobot/package.json]: OK!

File: javascript_nodejs/1.console-echobot/README.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+# To try this sample
+- Clone the repository
+    ```bash
+    git clone https://github.com/Microsoft/botbuilder-samples.git
+    ```
+- In a terminal, navigate to javascript_nodejs/1.console-echobot
+    ```bash
+    cd javascript_nodejs/1.console-echobot
+    ```
+    - Point to the myget feed
```
[the remark this answers is not in the document]
Comment (benbrown) [reply javascript_nodejs/1.console-echobot/README.md]: will fix. FYI, this is copy-pasted from the sample typescript project, so we should probably fix it there too before other people copy into their own projects.

File: javascript_nodejs/1.console-echobot/README.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
+This sample shows how to create a simple Echo bot that you can talk to from the terminal window..
+
+# To try this sample
+- Clone the repository
+    ```bash
+    git clone https://github.com/Microsoft/botbuilder-samples.git
```
[the remark this answers is not in the document]
Comment (benbrown) [reply javascript_nodejs/1.console-echobot/README.md]: noted.

**Comment by benbrown** [comment]
@vishwacsena @cleemullins I believe I have addressed all of your feedback. Thanks!

As I mentioned, some of the little grammar style issues @cleemullins pointed out were copy-pasted from [here](https://github.com/Microsoft/BotBuilder-Samples/tree/v4/javascript_ts/2.echobot-with-counter), so we should update there as well

**Pull request merged**